### PR TITLE
Adding rbac sdk path

### PR DESCRIPTION
**Pull request by ankush-cohere**
This PR introduces a new file, `compass_sdk/root_rbac.py`, which defines a `CompassRootClient` class for managing users, groups, and roles in a Compass system. The class provides methods for fetching, creating, and deleting users, groups, and roles, as well as managing role mappings.

The `CompassRootClient` class is initialised with a `compass_url` and a `root_user_token`, which are used to construct the base URL and authorisation headers for API requests. The class methods use the `requests` library to interact with the Compass API, performing various operations:

- `fetch_users` retrieves a list of users and returns a list of `UserFetchResponse` objects.
- `create_users` creates new users and returns a list of `UserCreateResponse` objects.
- `delete_users` deletes users by their names and returns a list of `UserDeleteResponse` objects.
- `fetch_groups` fetches a list of groups and returns a list of `GroupFetchResponse` objects.
- `create_groups` creates new groups and returns a list of `GroupCreateResponse` objects.
- `delete_groups` deletes groups by their names and returns a list of `GroupUserDeleteResponse` objects.
- `delete_user_group` removes a user from a group and returns a `GroupUserDeleteResponse` object.
- `fetch_roles` retrieves a list of roles and returns a list of `RoleFetchResponse` objects.
- `insert_roles` creates new roles and returns a list of `RoleCreateResponse` objects.
- `update_role` updates a role's policies and returns a `RoleCreateResponse` object.
- `delete_roles` deletes roles by their IDs and returns a list of `RoleDeleteResponse` objects.
- `insert_role_mappings` creates new role mappings and returns a list of `RoleMappingResponse` objects.
- `fetch_role_mappings` fetches a list of role mappings and returns a list of `RoleMappingResponse` objects.
- `delete_role_mappings` deletes role mappings for a specific role and group and returns a list of `RoleMappingDeleteResponse` objects.
- `create_root_user` creates a root user and returns a `UserCreateResponse` object.

Additionally, the PR adds a new file, `compass_sdk/types.py`, which defines various Pydantic models for the Compass system, including:

- `UserFetchResponse`, `UserCreateRequest`, `UserCreateResponse`, and `UserDeleteResponse` for user-related operations.
- `GroupFetchResponse`, `GroupCreateRequest`, `GroupCreateResponse`, and `GroupUserDeleteResponse` for group-related operations.
- `PolicyRequest`, `PolicyResponse`, `RoleFetchResponse`, `RoleCreateRequest`, `RoleCreateResponse`, and `RoleDeleteResponse` for role-related operations.
- `RoleMappingRequest`, `RoleMappingResponse`, and `RoleMappingDeleteResponse` for role mapping operations.

### CODE REVIEW

File: compass_sdk/types.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+
+class PolicyRequest(BaseModel):
+    indexes: List[str]
+    permission: str
+
+
+class PolicyResponse(BaseModel):
+    indexes: List[str]
+    permission: str
```
Comment (ankush-cohere) [inline compass_sdk/types.py:50]: Create a enum

File: compass_sdk/root_rbac.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    UserCreateRequest,
+    UserCreateResponse,
+    UserDeleteResponse,
+    UserFetchResponse,
+)
+
+
+def create_root_user(compass_url: str, tenancy_token: str) -> UserCreateResponse:
+    headers = {"Authorization": f"Bearer {tenancy_token}", "Content-Type": "application/json"}
+    response = requests.post(f"{compass_url}/security/admin/rbac/v1/root", headers=headers)
```
Comment (benrules3) [inline compass_sdk/root_rbac.py:28]: I think we probably want to include /api/ before security here, or user has to pass it in themselves.

Comment (benrules3) [reply compass_sdk/root_rbac.py]: So I do think nginx is expected in dev now that we're back to `run_docker.sh` - but I also agree this is a one-off, better to be handled with a script than the SDK

File: compass_sdk/rbac.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    RoleMappingRequest,
+    RoleMappingResponse,
+    UserCreateRequest,
+    UserCreateResponse,
+    UserDeleteResponse,
+    UserFetchResponse,
+)
+
+
+def create_root_user(compass_url: str, tenancy_token: str) -> UserCreateResponse:
```
[the remark this answers is not in the document]
Comment (ankush-cohere) [reply compass_sdk/rbac.py:28]: Yes, if you are okay with it, will remove it 👍

File: compass_sdk/compass.py [hunk 4]
```diff
@@ -473,6 +480,20 @@ def search(
             sleep_retry_seconds=1,
         )
 
+    def edit_group_authorization(self, *, index_name: str, group_auth_input: GroupAuthorizationInput):
```
[the remark this answers is not in the document]
Comment (ankush-cohere) [reply compass_sdk/compass.py:483]: All server side, don't make SDK smart is my rule of thumb

File: compass_sdk/compass.py [hunk 5]
```diff
@@ -115,6 +117,7 @@ def __init__(
             "add_context": "/api/v1/indexes/{index_name}/documents/add_context/{doc_id}",
             "refresh": "/api/v1/indexes/{index_name}/refresh",
             "push_documents": "/api/v2/indexes/{index_name}/documents",
+            "edit_group_authorization": "/v1/indexes/{index_name}/group_authorization",
```
[the remark this answers is not in the document]
Comment (ankush-cohere) [reply compass_sdk/compass.py:120]: My understanding for `api` is that nginx uses it. How does this work if we directly pass the compass url?

**Review by benrules3 — APPROVED** [review]

**Pull request merged**